### PR TITLE
Integration test optimization 115s -> 80s

### PR DESCRIPTION
**Pull request by JoshVanL**
Adds ParallelTest test helper. This can be used by integration tests as
an alternative to Go's `t.Parallel()` which requires spawning a new test
func which can make test result output noisy.

Updates iowriter to print exec output all at once at end of test
execution, rather than line by line.

Change placement and sentry integration processes to log at info level
rather than debug by default. This is to reduce the amount of logs
generated by the integration tests.

Updates the integration test suite to use the ParallelTest
rather than `t.Parallel()` to reduce test case number.

### CODE REVIEW

**Comment by ItalyPaleAle** [comment]
This should probably go into master and not 1.12 at this stage.

**Comment by artursouza** [comment]
> This should probably go into master and not 1.12 at this stage.

Yes. I agree.

File: tests/integration/framework/util/parallel.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	p := &ParallelTest{
+		fns: fns,
+	}
+	t.Cleanup(func() {
+		p.lock.Lock()
+		defer p.lock.Unlock()
+		t.Helper()
+
+		jobs := make(chan func(), len(p.fns))
+		for i := 0; i < 5; i++ {
```
Comment (filintod) [inline tests/integration/framework/util/parallel.go:42]: probably this `5` could be an env var or some other configurable variable

Comment (filintod) [reply tests/integration/framework/util/parallel.go:42]: other than that lgtm

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by daixiang0** [comment]
Start from https://github.com/dapr/dapr/actions/runs/6434413380/job/17473804904?pr=6982#step:7:15396, seems they are package list, do we really need them in log?

File: tests/integration/framework/util/parallel.go (hunk 1 above)
Comment (JoshVanL) [reply tests/integration/framework/util/parallel.go:42]: Done :+1:

**Comment by JoshVanL** [comment]
> Start from https://github.com/dapr/dapr/actions/runs/6434413380/job/17473804904?pr=6982#step:7:15396, seems they are package list, do we really need them in log?

That is the output from the integration tests running `go build`, but not sure why it is outputting that command here. Will need more investigation.

**Review by artursouza — APPROVED** [review]

**Pull request merged**